packages/gatsby/src/bootstrap/load-plugins/validate.ts

```diff
@@ -237,22 +239,56 @@ async function validatePluginsOptions(
         }
       } catch (error) {
         if (error instanceof Joi.ValidationError) {
+          // Show a small warning on unknown options rather than erroring
+          const validationWarnings = error.details.filter(
+            err => err.type === `object.unknown`
+          )
+          const validationErrors = error.details.filter(
+            err => err.type !== `object.unknown`
+          )
+
           // If rootDir and plugin.parentDir are the same, i.e. if this is a plugin a user configured in their gatsby-config.js (and not a sub-theme that added it), this will be ""
           // Otherwise, this will contain (and show) the relative path
           const configDir =
             (plugin.parentDir &&
               rootDir &&
               path.relative(rootDir, plugin.parentDir)) ||
             null
-          reporter.error({
-            id: `11331`,
-            context: {
-              configDir,
-              validationErrors: error.details,
-              pluginName: plugin.resolve,
-            },
-          })
-          errors++
+          if (validationErrors.length > 0) {
+            reporter.error({
+              id: `11331`,
+              context: {
+                configDir,
+                validationErrors,
+                pluginName: plugin.resolve,
+              },
+            })
+            errors++
+          }
+
+          if (validationWarnings.length > 0) {
+            reporter.warn(
+              stripIndent(`
+                Warning: there are unknown plugin options for "${
+                  plugin.resolve
+                }"${
+                configDir ? `, configured by ${configDir}` : ``
+              }: ${validationWarnings
+                .map(error => error.path.join(`.`))
+                .join(`, `)}
+                Please open an issue at ghub.io/${
+                  plugin.resolve
+                } if you believe this option is valid.
+              `)
+            )
+            trackCli(`UNKNOWN_PLUGIN_OPTION`, {
+              name: plugin.resolve,
+              valueString: validationWarnings
```

Comment:
This either needs adding to the definition for ITelemetryTagsPayload, or you need to use a key that _is_ in there. @jamo ?

Reply:
I've been fighting random test failures since yesterday with https://github.com/gatsbyjs/gatsby/pull/27949/files